### ISSUE TITLE
systemd specs should be in-repo

### ISSUE THREAD

**Issue by BenTheElder** [kubernetes/kubernetes#88832]
**What happened**:

https://github.com/kubernetes/kubernetes/pull/87585#pullrequestreview-369105857

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

/assign @justaugustus

**Comment by brenix** [comment]
I currently maintain the Archlinux AUR packages for [kubeadm](https://aur.archlinux.org/packages/kubeadm-bin/) and [kubelet](https://aur.archlinux.org/packages/kubelet-bin/), which relied on the default kubelet.service and kubeadm drop-in in the repo. It would be preferred to keep these in-repo so that other distributions can source the upstream default, rather than having to create it out of band. Thanks!

**Comment by neolit123** [comment]
> rather than having to create it out of band

currently these come from:
https://github.com/kubernetes/release/tree/master/cmd/kubepkg/templates/latest/deb/kubeadm

but matching the specs to a k8s tags/releases is probably better.

**Comment by BenTheElder** [comment]
another reason these should be versioned in the main repo https://github.com/kubernetes/release/pull/1352

**Comment by BenTheElder** [comment]
this is still true, for anyone not using the debs/rpms you either have to reinvent these or do quite some digging to turn up the upstream ones. consumers are not likely to be aware of kubernetes/release to begin with.

**Comment by BenTheElder** [comment]
At the very least we should probably leave a more obvious trail to find them. That might be less effort but still _somewhat_ helpful.

I still think we want them versioned with the binaries for other reasons though.

**Comment by BenTheElder** [comment]
I attempted to discuss this but I've been blocked on waiting for the new world of Kubernetes debian/rpm packaging to exist which has been in limbo for at least a year now, I really don't have the time / energy for this right now.

I still think it is wrong to have:
1) specs that are not versioned with kubernetes
    - this is blocking fixing the kubeadm crash looping kubelet experience
2) specs that are not easily found from the kubernetes project

**Comment by BenTheElder** [comment]
Kubelet service spec also doesn't have KillMode=process set when it probably should, but again that would technically be a breaking change and these files are unversioned.

Ever cluster tool inside and outside the project is just hard-forking the systemd files to fix things like this.

**Comment by tao12345666333** [comment]
Yes. Setting this option makes sense. And if it is version controlled, users will know about the change before upgrading.
